File: leetcode/1266/solution.go
/**
 * https://leetcode.com/problems/minimum-time-visiting-all-points/
 * 1266. Minimum Time Visiting All Points
 */
package main

func minTimeToVisitAllPoints(p [][]int) int {
	total := 0
	for i := 0; i < len(p)-1; i++ {
		total += max(abs(p[i][0], p[i+1][0]), abs(p[i][1], p[i+1][1]))
	}
	return total
}

func abs(a int, b int) int {
	if a > b {
		return a - b
	}
	return b - a
}

func max(a int, b int) int {
	if a > b {
		return a
	}
	return b
}

File: leetcode/1266/solution.rs
/**
 * https://leetcode.com/problems/minimum-time-visiting-all-points/
 * 1266. Minimum Time Visiting All Points
 */
impl Solution {
    pub fn min_time_to_visit_all_points(points: Vec<Vec<i32>>) -> i32 {
        points
            .windows(2)
            .map(|p| (((p[0][0] - p[1][0]).abs()).max((p[0][1] - p[1][1]).abs())))
            .sum()
    }
}
